### PR TITLE
Added `^` symbol before url paths in url.py files. 

### LINKED ISSUE #1216: Add ^ & $ in regex of urls for proper matching
' `^` ' sign should be introduced in the regular expressions of the url's, mentioned in the url files so that our search is always matched up correctly.

For example:  
If we have not added the initial caret character (i.e., 'hello/$'), Django would match any URL that ends with hello/, such as /my/site/hello/.
Also it will be more syntactically correct.

### PR DESCRIPTION
**Pull request by HargovindArora**
Fix #1216 .
URL files modified by adding `^` symbol before url patterns for correct path matching.

### CODE REVIEW

File: apps/challenges/urls.py [hunk 1]
```diff
@@ -4,38 +4,38 @@
...
 urlpatterns = [
 
-    url(r'challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/challenge$', views.challenge_list,
+    url(r'^challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/challenge$', views.challenge_list,
         name='get_challenge_list'),
-    url(r'challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/challenge/(?P<challenge_pk>[0-9]+)$',
+    url(r'^challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/challenge/(?P<challenge_pk>[0-9]+)$',
         views.challenge_detail, name='get_challenge_detail'),
-    url(r'challenge/(?P<challenge_pk>[0-9]+)/participant_team/(?P<participant_team_pk>[0-9]+)',
+    url(r'^challenge/(?P<challenge_pk>[0-9]+)/participant_team/(?P<participant_team_pk>[0-9]+)',
```
Comment (RishabhJain2018) [inline apps/challenges/urls.py]: Please add `$` at the end of url.

File: apps/challenges/urls.py [hunk 2]
```diff
@@ -4,38 +4,38 @@
...
         views.add_participant_team_to_challenge, name='add_participant_team_to_challenge'),
-    url(r'challenge/(?P<challenge_pk>[0-9]+)/disable',
+    url(r'^challenge/(?P<challenge_pk>[0-9]+)/disable',
```
Comment (RishabhJain2018) [inline apps/challenges/urls.py]: Same as above.

File: apps/challenges/urls.py [hunk 3]
```diff
@@ -4,38 +4,38 @@
...
         name='get_challenge_phase_list'),
-    url(r'challenge/(?P<challenge_pk>[0-9]+)/challenge_phase/(?P<pk>[0-9]+)$',
+    url(r'^challenge/(?P<challenge_pk>[0-9]+)/challenge_phase/(?P<pk>[0-9]+)$',
         views.challenge_phase_detail, name='get_challenge_phase_detail'),
     # `A-Za-z` because it accepts either of `all, future, past or present` in either case
-    url(r'challenge/(?P<challenge_time>[A-Za-z]+)$', views.get_all_challenges,
+    url(r'^challenge/(?P<challenge_time>[A-Za-z]+)$', views.get_all_challenges,
         name='get_all_challenges'),
-    url(r'challenge/(?P<pk>[0-9]+)/',
+    url(r'^challenge/(?P<pk>[0-9]+)/',
```
Comment (RishabhJain2018) [inline apps/challenges/urls.py]: Same as above.

File: apps/hosts/urls.py [hunk 4]
```diff
@@ -4,16 +4,16 @@
...
 urlpatterns = [
 
-    url(r'challenge_host_team/$', views.challenge_host_team_list, name='get_challenge_host_team_list'),
-    url(r'challenge_host_team/(?P<pk>[0-9]+)$', views.challenge_host_team_detail,
+    url(r'^challenge_host_team/$', views.challenge_host_team_list, name='get_challenge_host_team_list'),
+    url(r'^challenge_host_team/(?P<pk>[0-9]+)$', views.challenge_host_team_detail,
         name='get_challenge_host_team_details'),
-    url(r'create_challenge_host_team', views.create_challenge_host_team, name='create_challenge_host_team'),
-    url(r'challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/challenge_host$', views.challenge_host_list,
+    url(r'^create_challenge_host_team', views.create_challenge_host_team, name='create_challenge_host_team'),
```
Comment (RishabhJain2018) [inline apps/hosts/urls.py]: Same as above.

File: apps/jobs/urls.py [hunk 5]
```diff
@@ -3,16 +3,16 @@
 from . import views
 
 urlpatterns = [
-    url(r'challenge/(?P<challenge_pk>[0-9]+)/'
+    url(r'^challenge/(?P<challenge_pk>[0-9]+)/'
         r'challenge_phase/(?P<challenge_phase_pk>[0-9]+)/submission/(?P<submission_pk>[0-9]+)',
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

File: apps/jobs/urls.py [hunk 6]
```diff
@@ -3,16 +3,16 @@
...
         views.change_submission_data_and_visibility, name='change_submission_data_and_visibility'),
-    url(r'challenge/(?P<challenge_id>[0-9]+)/'
+    url(r'^challenge/(?P<challenge_id>[0-9]+)/'
         r'challenge_phase/(?P<challenge_phase_id>[0-9]+)/submission/',
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

File: apps/jobs/urls.py [hunk 7]
```diff
@@ -3,16 +3,16 @@
...
         views.challenge_submission, name='challenge_submission'),
-    url(r'(?P<challenge_pk>[0-9]+)/'
+    url(r'^(?P<challenge_pk>[0-9]+)/'
         r'phases/(?P<challenge_phase_pk>[0-9]+)/'
         r'remaining_submissions',
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

File: apps/jobs/urls.py [hunk 8]
```diff
@@ -3,16 +3,16 @@
...
         views.get_remaining_submissions, name='get_remaining_submissions'),
-    url(r'challenge_phase_split/(?P<challenge_phase_split_id>[0-9]+)/leaderboard/',
+    url(r'^challenge_phase_split/(?P<challenge_phase_split_id>[0-9]+)/leaderboard/',
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

**Comment by HargovindArora** [comment]
@RishabhJain2018 adding `$` at the end might make the urls to end immeditely and not allowing it to search for other instances like 'pks'. Please review it!

File: apps/accounts/urls.py [hunk 9]
```diff
@@ -4,6 +4,6 @@
 
 urlpatterns = [
 
-    url(r'^user/disable$', views.disable_user, name='disable_user'),
+    url(r'^user/disable/$', views.disable_user, name='disable_user'),
```
Comment (RishabhJain2018) [inline apps/accounts/urls.py]: Can you please brief me with the reason for using `/` before `$`. I might be missing some details here.

**Comment by HargovindArora** [comment]
@RishabhJain2018 the tests are failed. Should we append `$` sign in the end?

**Comment by modestlearner** [comment]
Can I fix this issue???

**Comment by RishabhJain2018** [comment]
@modestlearner Since @HargovindArora is already on this. Feel free to take up any other issue. :)

**Comment by RishabhJain2018** [comment]
@HargovindArora Yes, please add only the `$` sign at the end.

**Comment by HargovindArora** [comment]
My local branch got deleted so I create new PR #1238. Please refer to it.

**Pull request closed without merging**